File: redwood-yoga-java/src/main/java/io/github/orioncraftmc/meditate/interfaces/YogaBaselineFunction.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package io.github.orioncraftmc.meditate.interfaces;

import io.github.orioncraftmc.meditate.YogaNode;

public interface YogaBaselineFunction {
  /**
   * Return the baseline of the node in points. When no baseline function is set the baseline
   * default to the computed height of the node.
   */
  float baseline(YogaNode node, float width, float height);
}

File: redwood-yoga/src/commonMain/kotlin/app/cash/redwood/yoga/interfaces/YogaBaselineFunction.kt
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
package app.cash.redwood.yoga.interfaces

import app.cash.redwood.yoga.YogaNode

fun interface YogaBaselineFunction {
    /**
     * Return the baseline of the node in points. When no baseline function is set the baseline
     * default to the computed height of the node.
     */
    fun baseline(node: YogaNode?, width: Float, height: Float): Float
}
